utils/crypto/bls/signer/rpcsigner/client_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package rpcsigner
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+
+	"github.com/ava-labs/avalanchego/proto/pb/signer"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls/signer/localsigner"
+)
+
+type testClient struct{}
+
+var (
+	localSigner, _ = localsigner.New()
+
+	client = &Client{
+		client: &testClient{},
+		pk:     localSigner.PublicKey(),
+	}
+
+	validSignatureMsg        = []byte("valid")
+	uncompressedSignatureMsg = []byte("uncompressed")
+	emptySignatureMsg        = []byte("empty")
+	noSignatureMsg           = []byte("none")
+)
+
+func TestValidSignature(t *testing.T) {
+	sig, err := client.Sign(validSignatureMsg)
+	require.NoError(t, err)
+	bls.Verify(client.PublicKey(), sig, validSignatureMsg)
+}
+
+type test struct {
+	name string
+	msg  []byte
+	err  error
+}
+
+var tests = []test{
+	{
+		name: "Uncompressed",
+		msg:  uncompressedSignatureMsg,
+		err:  bls.ErrFailedSignatureDecompress,
+	},
+	{
+		name: "Empty",
+		msg:  emptySignatureMsg,
+		err:  ErrorEmptySignature,
+	},
+	{
+		name: "None",
+		msg:  noSignatureMsg,
+		err:  ErrorEmptySignature,
+	},
+}
+
+func TestInvalidSignature(t *testing.T) {
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			sig, err := client.Sign(test.msg)
+			require.Nil(t, sig)
+			require.EqualError(t, err, test.err.Error())
+		})
+	}
+}
+
+func TestValidPOPSignature(t *testing.T) {
+	sig, err := client.SignProofOfPossession(validSignatureMsg)
+	require.NoError(t, err)
+	bls.VerifyProofOfPossession(client.PublicKey(), sig, validSignatureMsg)
+}
+
+func TestInvalidPOPSignature(t *testing.T) {
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			sig, err := client.SignProofOfPossession(test.msg)
+			require.Nil(t, sig)
+			require.EqualError(t, err, test.err.Error())
+		})
+	}
+}
+
+func (*testClient) PublicKey(_ context.Context, _ *signer.PublicKeyRequest, _ ...grpc.CallOption) (*signer.PublicKeyResponse, error) {
+	return nil, nil
```

Comment:
I think this is supposed to return the pk instead of nil - my guess is that this isn't caught because of the missing assertions